packages/static-wado-webserver/lib/routes/server/studies.mjs

```diff
@@ -27,10 +28,37 @@ import renderedMap from "../../controllers/server/renderedMap.mjs";
  * @param {*} routerExpress root entry point for studies routes.
  * @param {*} params
  * @param {*} dir static files directory path
+ * @param {*} hashStudyUidPath change studies folder structure to path and subpath before studyUID
  */
-export default function setRoutes(routerExpress, params, dir) {
+export default function setRoutes(
+  routerExpress,
+  params,
+  dir,
+  hashStudyUidPath
+) {
   createStudyDirectories(dir);
 
+  routerExpress.use("/", (req, res, next) => {
+    if (hashStudyUidPath) {
+      const originalPath = req.path;
+      const studyUID = originalPath.match(/\/studies\/([^/]+)/)?.[1]; // get UID only
+
+      if (studyUID) {
+        const { path: hashPath = "", subpath: hashSubpath = "" } =
+          getStudyUIDPathAndSubPath(studyUID);
+        const hashPrefix = `${hashPath}/${hashSubpath}`;
+        const newPath = originalPath.replace(
+          `/studies/${studyUID}`,
+          `/studies/${hashPrefix}/${studyUID}`
+        );
+
+        req.hashedPath = newPath;
```

Comment:
The redirections fail if you don't have hashStudyUidPath on.  Please always create the "hashedPath" value, just using the original value when it isn't present.